File: swiggy-backend/src/main/java/com/swiggy/swiggy_backend/service/CartService.java
package com.swiggy.swiggy_backend.service;

import com.swiggy.swiggy_backend.dto.AddToCartRequest;
import com.swiggy.swiggy_backend.dto.CartResponse;


public interface CartService {

    CartResponse addToCart(AddToCartRequest request);

    CartResponse getCart();

    CartResponse updateCartItem(Long cartItemId, Integer quantity);

    void removeCartItem(Long cartItemId);

    void clearCart();

}

File: swiggy-frontend/src/services/cartService.js
import api from "./api";

export const addToCart = (menuItemId, quantity = 1) => {
    return api.post("/cart/items", {
        menuItemId,
        quantity,
    });
};

export const getCart = () => {
    return api.get("/cart");
};

export const updateCartItem = (cartItemId, quantity) => {
    return api.put(`/cart/items/${cartItemId}`, null, {
        params: {
            quantity,
        },
    });
};

export const removeCartItem = (cartItemId) => {
    return api.delete(`/cart/items/${cartItemId}`);
};

export const clearCart = () => {
    return api.delete("/cart/clear");
};